sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -271,6 +276,14 @@ private String ensureSlashPrefixed(String path) {
             return contentObservable
                        .flatMap(content -> {
                            try {
+                               //Adding transport client request timeline to diagnostics
+                               ReactorNettyRequestRecord reactorNettyRequestRecord = httpResponseTuple.getRight();
+                               if (reactorNettyRequestRecord != null) {
+                                   reactorNettyRequestRecord.setTimeCompleted(OffsetDateTime.now());
```

Comment:
shouldn't we capture the timeline inside the response instead of the request?
Why are capturing the timeline in the request not the response?

Reply:
this is just a question, please provide the reason why this should go to request not response

Reply:
All the diagnostic we capture in request , as it flows to different pipeline, retries , capturing different data , and once we have the response then we transfer the diagnostic from request to response.

Reply:
Moved ReactorNettyRequestRecord into HttpRequest